src/code/z_scene_table.c

```diff
@@ -30,784 +30,1682 @@
-    ENTRANCE(0x56, 0x02, 0, 1, 0x12, 0x12), ENTRANCE(0x56, 0x02, 0, 1, 0x02, 0x02),
-    ENTRANCE(0x56, 0x02, 0, 1, 0x12, 0x12), ENTRANCE(0x56, 0x02, 0, 1, 0x02, 0x02),
-    ENTRANCE(0x57, 0x08, 0, 1, 0x02, 0x02), ENTRANCE(0x57, 0x08, 0, 1, 0x02, 0x02),
-    ENTRANCE(0x57, 0x08, 0, 1, 0x02, 0x02), ENTRANCE(0x57, 0x08, 0, 1, 0x02, 0x02),
-    ENTRANCE(0x56, 0x03, 0, 1, 0x03, 0x03), ENTRANCE(0x56, 0x03, 0, 1, 0x03, 0x03),
-    ENTRANCE(0x56, 0x03, 0, 1, 0x03, 0x03), ENTRANCE(0x56, 0x03, 0, 1, 0x03, 0x03),
-    ENTRANCE(0x57, 0x09, 0, 1, 0x03, 0x03), ENTRANCE(0x57, 0x09, 0, 1, 0x03, 0x03),
-    ENTRANCE(0x57, 0x09, 0, 1, 0x03, 0x03), ENTRANCE(0x57, 0x09, 0, 1, 0x03, 0x03),
-    ENTRANCE(0x5C, 0x08, 0, 1, 0x03, 0x03), ENTRANCE(0x5C, 0x08, 0, 1, 0x03, 0x03),
-    ENTRANCE(0x5C, 0x08, 0, 1, 0x03, 0x03), ENTRANCE(0x5C, 0x08, 0, 1, 0x03, 0x03),
+    ENTRANCE(SCENE_YDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_YDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_YDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_YDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_DDAN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MEN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MEN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MEN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MEN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MORIBOSSROOM, 0, true, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MORIBOSSROOM, 0, true, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MORIBOSSROOM, 0, true, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MORIBOSSROOM, 0, true, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MIZUSIN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MIZUSIN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MIZUSIN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(SCENE_MIZUSIN, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(0x6E, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(0x6E, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(0x6E, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
+    ENTRANCE(0x6E, 0, false, true, TRANS_TYPE_FADE_BLACK, TRANS_TYPE_FADE_BLACK),
```

Comment:
Perhaps others already already know, but do you know what's going on with scene `0x6E`? Does this enum value appear anywhere else in code?

Also as a tangent, just a random thought I had. Is there a reason `sceneNum` is used instead of `sceneId` throughout code? `Num` often means total with our style of naming afaik

Reply:
its referencing a scene that doesnt exist in the game. everything about it got removed and all we have is this reference to its scene number. Not sure if we have any knowledge of that the scene actually was, but there isnt afaik.

I had this same thought recently. Pretty sure its just rom hacking carry overs and id would probably make more sense

Reply:
Do you think it's worth adding that removed scene to the enum list? I'm wondering if someone else comes in and sees the number `0x6E` and wonders the same thing. It would be easier to search the repo for `SCENE_REMOVED` to see that this is the only place it's used.

I guess the other thing to be mindful of is when retail support is incorporated and some of the debug scenes are also removed for those version. So maybe that's not the best name.

Reply:
with how the scene table works (DEFINE_SCENE) an enum entry cant really be added without also defining other stuff that makes it not match anymore. at least as far as im aware

Reply:
and i also dont see any benefit from having it anyway, as it references something that doesnt exist literally anywhere else

Reply:
not having it part of the enum technically breaks shiftability, since adding another scene will change where those entrances go.

Reply:
realized that we do this to, for example, add a "max" entry for other enums that get generated by tables
```
typedef enum {
    #include "tables/actor_table.h"
    /* 0x0192 */ ACTOR_ID_MAX // originally "ACTOR_DLF_MAX"
} ActorID;
```
so we could do that here. just not sure on a name.
it could be named as MAX since its the max in every other sense, but happens to get used as an entry in the entrance table. or it could be named after the fact that its removed/not used

Reply:
-1 on adding this deleted (?) scene to the scene enum
if we really want to document it we could have a `#define` for that particular value (and I don't think we should care since it's only a set of entries in the entrance table), but it shouldn't be part of the actual enum imo

Reply:
I agree to a point, but also think mzx has a point, theat when modders add a new scene to the end of the list this entrance would have an entry that uses its value. Itd be pretty confusing